datasets/flwr_datasets/resplitter/divide_resplitter.py

```diff
@@ -0,0 +1,286 @@
+                new_splits.extend(list(new_splits_dict.keys()))
+            all_splits = dataset_splits + new_splits
+            for used_split in self._multiple_splits_config.keys():
+                all_splits.pop(all_splits.index(used_split))
+        else:
+            raise ValueError("Incorrect type of config.")
+
+        duplicates = [
+            item for item, count in collections.Counter(all_splits).items() if count > 1
+        ]
+        if duplicates:
+            raise ValueError(
+                "The specified values of the new splits in "
+                "`divide_config` are duplicated with the split names of "
+                "the datasets. "
+                "Please specify unique values for each new split."
+            )
+
+    def _check_size_values(self, dataset: DatasetDict) -> None:
+        # It should be called after the `divide_config` is in the multiple-splits format
+        assert self._multiple_splits_config is not None
+        for split_from, new_split_dict in self._multiple_splits_config.items():
+            if all(isinstance(x, float) for x in new_split_dict.values()):
+                if not all(0 < x <= 1 for x in new_split_dict.values()):
+                    raise ValueError(
+                        "All fractions in `divide_config` must be greater than 0 and "
+                        "smaller or equal to 1."
+                    )
+                if sum(new_split_dict.values()) > 1.0:
+                    raise ValueError(
+                        "The sum of the fractions in `divide_config` must be smaller "
+                        "than 1.0."
+                    )
+
+            elif all(isinstance(x, int) for x in new_split_dict.values()):
+                dataset_len = len(dataset[split_from])
+                len_from_divide_resplit = sum(new_split_dict.values())
+                if len_from_divide_resplit > dataset_len:
+                    raise ValueError(
+                        f"The sum of the sample numbers in `divide_config` must be "
+                        f"smaller than the split size. This is not the case for "
+                        f"{split_from} split which is of length {dataset_len} and the "
+                        f"sum in config is {len_from_divide_resplit}."
+                    )
+            else:
+                raise TypeError(
+                    "The values in `divide_config` must be either ints or floats. "
+                    "The mix of them or other types are not allowed."
+                )
+
+    def _warn_on_potential_misuse_of_divide_split(self) -> None:
+        if self._config_type == "multiple-splits" and self._divide_split is not None:
+            warnings.warn(
+                "The `divide_split` was specified but the multiple split "
+                "configuration was given. The `divide_split` will be "
+                "ignored.",
+                stacklevel=1,
+            )
+
+
+def _determine_config_type(
```

Comment:
I wonder if we could enhance this function to check not only if `divide_config` is single-split or multi-split but also if it's "valid" (i.e. to check that actually a single-split `divide_config` is of type `Union[Dict[str,int], Dict[str, float]]` and nothing else.

Reply:
I noted afterwards you do this in `_check_size_values()` but I wonder if we should better do this check earlier. Let's keep it as is if it's better to do this checks each time upon `__call__`.

Reply:
Ok, I'll add a separate check on the type validity.

Reply:
Done

Reply:
Also, a thing we can consider in the future is adding the external type checker e.g. `from typeguard import typechecked` and then 
```@typechecked
def whatever(param: annotation) -> sth:
```
or e.g.`pydantic`